subgraphs/stakedao/src/Oracle/index.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  dataSource,
+  log,
+} from "@graphprotocol/graph-ts";
+
+import * as constants from "./common/constants";
+import { getCurvePriceUsdc } from "./routers/CurveRouter";
+import { getPriceUsdc as getPriceUsdcUniswap } from "./routers/UniswapRouter";
+import { getPriceUsdc as getPriceUsdcSushiswap } from "./routers/SushiSwapRouter";
+
+export function getUsdPricePerToken(tokenAddr: Address): BigDecimal[] {
+  let network = dataSource.network();
+  let decimals = BigInt.fromI32(10)
+    .pow(BigInt.fromI32(6).toI32() as u8)
+    .toBigDecimal();
+
+  // ? 1. Curve Router
+  let curvePrice = getCurvePriceUsdc(tokenAddr, network);
+  if (curvePrice.notEqual(constants.BIGDECIMAL_ZERO)) {
+    log.warning("[getCurvePriceUsdc] tokenAddress: {}, curvePrice: {}", [
+      tokenAddr.toHexString(),
+      curvePrice.div(decimals).toString(),
+    ]);
+
+    return [curvePrice, decimals];
+  }
+
+  //? 2. Uniswap Router
+  let uniswapPrice = getPriceUsdcUniswap(tokenAddr, network);
+  if (uniswapPrice.notEqual(constants.BIGDECIMAL_ZERO)) {
+    log.warning("[getPriceUsdcUniswap] tokenAddress: {}, uniswapPrice: {}", [
+      tokenAddr.toHexString(),
+      uniswapPrice.div(decimals).toString(),
+    ]);
+
+    return [uniswapPrice, decimals];
+  }
+
+  //? 3. SushiSwap Router
+  let sushiswapPrice = getPriceUsdcSushiswap(tokenAddr, network);
+  if (sushiswapPrice.notEqual(constants.BIGDECIMAL_ZERO)) {
+    log.warning(
+      "[getPriceUsdcSushiswap] tokenAddress: {}, sushiswapPrice: {}",
+      [tokenAddr.toHexString(), sushiswapPrice.div(decimals).toString()]
+    );
+
+    return [sushiswapPrice, decimals];
+  }
+
+  // TODO: WIP - ChainLink Router
+  // //? 4. ChainLink

```

Comment:
Should always go with chainlink first

Reply:
The ordering is completely dependent on the subgraphs requirement, my subgraph(Stake DAO) is completely dependent on Curve Router and only the Curve router works for me, that's why it's number one for me. 

Ideally, the subgraph developers should change the ordering depending on their subgraph, so that it could index a little faster and reduce unnecessary JSON RPC Calls.